### PR TITLE
Watch App

### PR DESCRIPTION
**Pull request by algesten**
Need to update submodules again once we merge https://github.com/mattrubin/Base32/pull/1 and https://github.com/mattrubin/OneTimePassword/pull/94

### CODE REVIEW

**Comment by algesten** [comment]
I don't even understand what travis is complaining about :/

**Comment by beaucollins** [comment]
I think travis is missing the watch simulator so it's not able to run build the project.

```
Testing failed:
	Target specifies product type 'com.apple.product-type.watchkit2-extension', but there's no such product type for the 'iphonesimulator' platform
	Unable to run command 'PBXCp Base32.framework' - this target might include its own product.
	Unable to run command 'PBXCp OneTimePassword.framework' - this target might include its own product.
** TEST FAILED **
```

**Comment by algesten** [comment]
wonder whether it can be that the project settings say "latest watchos 3.1" while the travis conf says

```
matrix:
  include:
    - osx_image: xcode7.3
      xcode_sdk: iphonesimulator9.3
```

watchos 3 is only iOS 10, so it may be that a 9.3 simulator can never run this.

**Comment by algesten** [comment]
but then to contradict myself – i think i built everything to be watchos 9 compatible, only using watchos 2 API since the rest of the project is thereabouts.

File: Authenticator.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -640,13 +959,216 @@
...
+				MTL_ENABLE_DEBUG_INFO = YES;
+				PRODUCT_BUNDLE_IDENTIFIER = me.mattrubin.authenticator.dev.watchkitapp;
+				PRODUCT_NAME = "$(TARGET_NAME)";
+				SDKROOT = watchos;
+				SKIP_INSTALL = YES;
+				SWIFT_ACTIVE_COMPILATION_CONDITIONS = DEBUG;
+				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 3.0;
+				TARGETED_DEVICE_FAMILY = 4;
+				WATCHOS_DEPLOYMENT_TARGET = 3.0;
```
Comment (beaucollins) [inline Authenticator.xcodeproj/project.pbxproj]: This might be why travis can't build it, try with 2.0.

Comment (algesten) [inline Authenticator.xcodeproj/project.pbxproj]: cool! i give it a go

Comment (algesten) [inline Authenticator.xcodeproj/project.pbxproj]: still no luck.

```
	Target specifies product type 'com.apple.product-type.watchkit2-extension', but there's no such product type for the 'iphonesimulator' platform
```

the error message is a bit curious. it's like it's trying to run the watch extension on a iphone simulator, which won't work.

**Comment by algesten** [comment]
I've rebased and removed unnecessary commits. Do you want to the keychain reset and toURLWithSecret() as separate commits, or should I squash it to one "watch app" commit?

**Pull request closed without merging**